Algorithm/jumpsearch.cpp

```diff
@@ -0,0 +1,57 @@
+#include <bits/stdc++.h> 
+using namespace std; 
+  
+int jumpSearch(int arr[], int x, int n) 
+{ 
+    // Finding block size to be jumped 
+    int step = sqrt(n); 
+  
+    // Finding the block where element is 
+    // present (if it is present) 
+    int prev = 0; 
+    while (arr[min(step, n)-1] < x) 
+    { 
+        prev = step; 
+        step += sqrt(n); 
+        if (prev >= n) 
+            return -1; 
+    } 
+  
+    // Doing a linear search for x in block 
+    // beginning with prev. 
+    while (arr[prev] < x) 
+    { 
+        prev++; 
+  
+        // If we reached next block or end of 
+        // array, element is not present. 
+        if (prev == min(step, n)) 
+            return -1; 
+    } 
+    // If element is found 
+    if (arr[prev] == x) 
+        return prev; 
+  
+    return -1; 
+} 
+  
+// Driver program to test function 
+int main() 
+{ 
+    int n;
+    cout<<"enter no. of elements in array";
+    cin>>n;
+    int arr[n];
```

Comment:
`n` should be a constant

Reply:
we can take n to be a variable to give user the independence of no. of elements

Reply:
You can create heap memory with `new/delete`